accord-core/src/main/java/accord/utils/RandomSource.java

```diff
@@ -266,4 +345,15 @@ public double nextGaussian()
             }
         };
     }
+
+    class Impl // once jdk8 is dropped can have private functions and replace this class
+    {
+        private static void checkNextArray(Object array, int realLength, int offset, int length)
```

Comment:
This is a confusingly named function, particularly since the "array" is passed as a parameter. Perhaps move to `Invariants` and call it `indexInBounds`?

Reply:
removed and pushed to `Invariants`